astarte-device-sdk/AstarteDeviceSDK.cpp

```diff
@@ -373,3 +373,33 @@ AstarteDeviceSDK::ConnectionStatus AstarteDeviceSDK::connectionStatus() const
 
     return static_cast<AstarteDeviceSDK::ConnectionStatus>(m_astarteTransport->connectionStatus());
 }
+
+template <typename T>
+bool AstarteDeviceSDK::sendData(const QByteArray &interface, const QByteArray &path, const QList<T> &value, const QDateTime &timestamp, const QVariantHash &metadata)
+{
+    QList<QVariant> value2;
+    value2.reserve(value.length());
+
+    for (auto i: value) {
+        value2.append(QVariant(i));
+    }
+
+    return sendData(interface,path,value2,timestamp,metadata);
+}
+
+template bool AstarteDeviceSDK::sendData(const QByteArray &interface, const QByteArray &path, const QList<int> &value, const QDateTime &timestamp, const QVariantHash &metadata);
+template bool AstarteDeviceSDK::sendData(const QByteArray &interface, const QByteArray &path, const QList<double> &value, const QDateTime &timestamp, const QVariantHash &metadata);
+template bool AstarteDeviceSDK::sendData(const QByteArray &interface, const QByteArray &path, const QList<bool> &value, const QDateTime &timestamp, const QVariantHash &metadata);
+template bool AstarteDeviceSDK::sendData(const QByteArray &interface, const QByteArray &path, const QList<QDateTime> &value, const QDateTime &timestamp, const QVariantHash &metadata);
+template bool AstarteDeviceSDK::sendData(const QByteArray &interface, const QByteArray &path, const QList<QString> &value, const QDateTime &timestamp, const QVariantHash &metadata);
+
+
+bool AstarteDeviceSDK::sendData(const QByteArray &interface, const QByteArray &path, const QList<QVariant> &value, const QDateTime &timestamp, const QVariantHash &metadata)
```

Comment:
I think this function is pretty useless, let's remove it.
QVariant has a QList<QVariant>.

Reply:
It's used to call the correct sendData in AstarteGenericProducer.cpp